src/code/z_parameter.c

```diff
@@ -2264,7 +2264,7 @@ void Magic_Update(PlayState* play) {
             // Add magic until magicFillTarget is reached
             gSaveContext.save.playerData.magic += 0x10;
 
-            if ((gSaveContext.gameMode == 0) && (gSaveContext.sceneSetupIndex < 4)) {
+            if ((gSaveContext.gameMode == 0) && (gSaveContext.sceneLayer < 4)) {
```

Comment:
This line seems to have not been updated from OoT, where there are 4 non-cutscene layers instead of MM's 1. See here:
https://github.com/zeldaret/oot/blob/4f65d08eb5826c9c3520fd398b2e76ab09168193/src/code/z_parameter.c#L2442

Might be worth a comment